### PR TITLE
[BEAM-12976] Enable projection pushdown in runners

### PR DESCRIPTION
**Pull request by ibzib**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by ibzib** [comment]
Run PostCommit_Java_Dataflow

**Comment by ibzib** [comment]
Run PostCommit_Java_Dataflow

**Comment by ibzib** [comment]
R: @apilloud

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java [hunk 1]
```diff
@@ -81,6 +82,11 @@ public PipelineResult run(Pipeline pipeline) {
       SplittableParDo.convertReadBasedSplittableDoFnsToPrimitiveReadsIfNecessary(pipeline);
     }
 
+    if (!options.isStreaming()
```
Comment (apilloud) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java:85]: What is the risk for Streaming pipelines?

**Review by apilloud — APPROVED** [review]
LGTM

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java (hunk 1 above)
Comment (ibzib) [reply runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkRunner.java:85]: @angoenka mentioned that updates for streaming pipelines might fail if we change the pipeline graph.

**Comment by ibzib** [comment]
run java precommit

**Comment by ibzib** [comment]
portable python precommit is broken https://issues.apache.org/jira/browse/BEAM-14122

**Comment by ibzib** [comment]
Looks like spanner test failure was just a flake. The optimizer didn't run on it, so it's unrelated.

**Pull request merged**